### ISSUE TITLE
Bug Report - Setting "Disable preview" doesn’t seem to work as expected

### ISSUE THREAD

**Issue by AndMel87** [fgl27/SmartTwitchTV#63]
Setting «Disable preview when multi-stream is enabled» from the Preview thumbnail player settings doesn’t seem to work as expected. The preview window still tries to load, and throws error message «Player unable to play due to player error. Error code:4001».

**Comment by AndMel87** [comment]
Apk: 3.0.325 
web: July 22 2021 
Webview: 91.0.4472.164 
Device: TCL - Percee TV
Sdk: 26

**Comment by fgl27** [comment]
To resolve this I need more info from the app take a picture or screenshot the setting option Blocked codecs and the about of the app and share here

As on bellow images yours will look different


![Screenshot from 2021-07-31 09-03-46](https://user-images.githubusercontent.com/13755892/127739307-7ccf4d32-f956-42e2-94fe-8df7d9dfb353.png)

![Screenshot from 2021-07-31 09-02-17](https://user-images.githubusercontent.com/13755892/127739308-d33366c5-4ef4-4787-8454-0239c6b4fe1c.png)

**Comment by AndMel87** [comment]
Attached:

![about](https://user-images.githubusercontent.com/47888106/127743788-09f87c57-9e46-4d39-a791-a5bc87910452.jpg)
![codecs](https://user-images.githubusercontent.com/47888106/127743790-48c842ba-52ca-44d5-8014-cf745ecb46d8.jpg)

**Comment by fgl27** [comment]
in relation to the settings option, "multi-stream" is the for way multistream, something that yours device doesn't support and you see the below warning when you try to use it

That option will only prevent the preview from show when multistream is enabled

# Multistream warning

![Screenshot from 2021-08-01 06-16-45](https://user-images.githubusercontent.com/13755892/127766311-ce8c1c54-0a44-43d9-9c05-6e10ffd5de5b.png)

**Comment by fgl27** [comment]
Regarding the error 4001 there reason is "Caused by a decoder initialization failure", because yours device only supports two players at the same time.

The app know that you can only use two and ca n prevent the preview from show, I just need to add a new check for it.

**Comment by fgl27** [comment]
I have added the bellow warning, it will show only once the first time it detects that you are trying to start more player then what the device supports

![Screenshot from 2021-08-01 07-34-16](https://user-images.githubusercontent.com/13755892/127767865-f2fcef6b-9df4-47b1-b6a1-3f17d9a13eed.png)

**Comment by hunter1749** [comment]
> in relation to the settings option, "multi-stream" is the for way multistream, something that yours device doesn't support and you see the below warning when you try to use it
> 
> That option will only prevent the preview from show when multistream is enabled
> 
> # Multistream warning
> ![Screenshot from 2021-08-01 06-16-45](https://user-images.githubusercontent.com/13755892/127766311-ce8c1c54-0a44-43d9-9c05-6e10ffd5de5b.png)

Is there an option that would disable the "Your device only supports 2 codec instances" message to show like that? For me it now shows almost every time I just normally open a new stream (so even if there aren't multiple instances playing, but maybe it's still detecting that the preview thumbnail is being played at the same time and it shows the message on stream because of that?). This started happening about a week ago.

**Comment by fgl27** [comment]
**Are you sure you are see that warning? on the middle of the screen**

That exactly string can only show if you try to open multistream, by clicking on the player controls or using the KEY_MEDIA_REWIND... the end of the string is "4-way multistream" only gets used here:

https://github.com/fgl27/SmartTwitchTV/blob/ca720dcd6491e90bfa49fb041685d5c71e8d5511/app/specific/PlayEtc.js#L2753


The other warning ends with "the preview", only shows once on the situation of the image of my previous post that I re-add below, then will only show again if you close the app and try again to use the preview and PP mode at same time but only once never again until the app is closed.

![Screenshot from 2021-08-01 07-42-37](https://user-images.githubusercontent.com/13755892/127767952-24e06929-8e0d-4212-9dd1-c92f03abcfd6.png)

**Comment by fgl27** [comment]
![Screenshot_20210804-122856_SmartTV_Client_for_Twitch](https://user-images.githubusercontent.com/13755892/128209266-d8bd4cff-6468-46b9-bc8a-7d8a418c4a1e.png)

Is possible to add more options to that settings options, but I would like to know if there is warnings shown when it shouldn't, so I can prevent it.

**Comment by hunter1749** [comment]
Yes, I see it in the middle of the screen when I click on one of the livestreams. I can create another issue when I have time if needed,

**Comment by fgl27** [comment]
click where?

**Comment by hunter1749** [comment]
On the user menu (where livestreams of my followed channels are) I click on a livestream that I want to watch and the message appears in the middle of the screen when I select a livestream. Then that message disappears after a few seconds.

**Comment by fgl27** [comment]
I can't understand how that show

open a separated issue and make a video of the issue please
